content/docs/grammar-language.md

```diff
@@ -312,25 +312,37 @@ Element:
     '}';
 ```
 The only difference between `RootElement` and `Element` is that the former has the boolean property `isPublic`. 
-We can refactor the grammar so that only `Element` is present in the grammar with a *Guard* that will determine what syntax should be used by the parser:
+We can refactor the grammar so that only `Element` is present in the grammar with a *guard condition* that will determine which concrete syntax should be used by the parser:
 ```
 entry Model:
     element+=Element<true>;
 
 Element<isRoot>:
-	<isRoot> isPublic+='public'? 
-	'element' name=ID '{'
-		elements+=Element<false>*
-	'}'
-	|
+	(<isRoot> isPublic?='public')? 
 	'element' name=ID '{'
 		elements+=Element<false>*
 	'}';
 ```
-`Element` has the guard `isRoot`, which will determine which of the two alternatives should be matched by the parser. 
-If `isRoot` is set to `true`, then the first alternative is matched. Otherwise, the second alternative will apply. 
+`Element` has the guard `isRoot`, which will determine whether the optional group containing the `isPublic` property is allowed to be parsed.
+
 The *entry rule* `Model` sets the value of `isRoot` to `true` with `element+=Element<true>`, while `isRoot` is set to `false` inside of the `Element<isRoot>` parser rule with `elements+=Element<false>`.
 
+In general, a guard condition on a group decides whether the parser is allowed to parse the group or not depending on the result of the evaluated condition. Logical operations can be applied, such as `&` (and), `|` (or) and `!` (not) to fine-tune the exact conditions in which the group is supposed to be parsed.
+
+Additionally, guard conditions can also be used inside of alternatives. See the following example:
```

Comment:
Just to be sure I am clear about the syntax, let's consider:
```
Element<isRoot>:
	(<isRoot> 'root' | 'element') name=ID '{'
		elements+=Element<false>*
	'}';
```

Would this be equivalent to the example you give in the documentation?

Reply:
It wouldn't there's a slight difference:

`(<isRoot> 'root' | 'element')` __always__ allows to parse `element`, while it allows `root` if `isRoot=true`, while `(<isRoot> 'root' | <!isRoot> 'element')` makes it __mutually exclusive__. Only one is possible at any given moment.